internal/testutil/test_setup.go

```diff
@@ -42,13 +44,29 @@ func SmokeTest(t *testing.T) {
 func UnitTest(t *testing.T) {
 	t.Helper()
 	c := config.New()
-	c.SetManageBinariesAutomatically(false)
 	c.SetToken("00000000-0000-0000-0000-000000000001")
 	c.SetTrustedFolderFeatureEnabled(false)
 	config.SetCurrentConfig(c)
 	CLIDownloadLockFileCleanUp(t)
 	notification.DisposeListener()
-	t.Cleanup(func() { notification.DisposeListener() })
+	t.Cleanup(func() {
+		notification.DisposeListener()
+		cleanupFakeCliFile(c)
+	})
+}
+
+func cleanupFakeCliFile(c *config.Config) {
```

Comment:
Suggestion: How about moving this logic to FakeInstaller, so that creating the fake file and deleting it are close by.

Reply:
@bastiandoetsch and I decided to postpone this and unblock this PR.

Reply:
Good idea, to move it. But, as discussed, let's investigate tomorrow, if we can move FakeInstaller instead.